lib/Semantics/check-select-stmt.cpp

```diff
@@ -0,0 +1,126 @@
+//===-- lib/semantics/check-select-stmt.cpp -------------------------------===//
```

Comment:
This is not a good name for this file.  Fortran has three different `SELECT` statements and this file only checks one of the three.

Reply:
How about `check-select-rank.[cpp|h]`?